automatic/freecad/tools/chocolateyInstall.ps1

```diff
@@ -1,17 +1,62 @@
-﻿$ErrorActionPreference = 'Stop';
+$ErrorActionPreference = 'Stop';
+
+$toolsDir = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
+
+# Checking for Package Parameters
+$pp = Get-PackageParameters
+if (!$pp['UnzipLocation']) { $pp['UnzipLocation'] = "$toolsDir" }
 
 $packageArgs = @{
   packageName    = $env:ChocolateyPackageName
-  fileType       = 'exe'
-  url            = 'https://github.com/FreeCAD/FreeCAD/releases/download/0.18.3/FreeCAD-0.18.16131.3129ae4-WIN-x32-installer.exe'
-  url64          = 'https://github.com/FreeCAD/FreeCAD/releases/download/0.18.3/FreeCAD-0.18.16131.3129ae4-WIN-x64-installer.exe'
+  fileType       = ''
+  url            = ''
+  url64          = ''
+  UnzipLocation	 = $pp.UnzipLocation
   softwareName   = 'FreeCAD*'
-  checksum       = 'ef5d4108329c71b4b5b60fbe2fbc49acef86694942e83765a4863010eabe0d79'
-  checksumType   = 'sha256'
-  checksum64     = 'c521b8ca5dd9ed287bc3afbe57aa39f4dea4ca89e13fb58c3258d3792f86d48f'
-  checksumType64 = 'sha256'
+  checksum       = ''
+  checksumType   = ''
+  checksum64     = ''
+  checksumType64 = ''
   silentArgs     = '/S'
   validExitCodes = @(0)
 }
 
+$fileName = @{$true=($packageArgs.url);$false=($packageArgs.url64)}[ ((Get-OSBitness) -eq 32) ]
+$filename = $fileName -split('/'); $filename = ( $filename[-1] ); $filename = ( $filename -replace( "\.$($packageArgs.fileType)", '' ) )
+
+# Checking for Package Parameters
+if (!$pp['WorkingDirectory']) { $pp['WorkingDirectory'] = $pp.UnzipLocation+"\$filename" }
+if (!$pp['TargetPath']) { $pp['TargetPath'] = $pp['WorkingDirectory']+"\bin\${env:ChocolateyPackageTitle}.exe" }
+if (!$pp['IconLocation']) { $pp['IconLocation'] = $pp['TargetPath'] }
+if (!$pp['Arguments']) { $pp['Arguments'] = "" }
+if (!$pp['ShortcutFilePath']) { $pp['ShortcutFilePath'] = ( [Environment]::GetFolderPath('Desktop') )+"\${env:ChocolateyPackageTitle}.lnk" }
+if (!$pp['Shortcut']) { $pp['Shortcut'] = $true }
+if (!$pp['WindowStyle']) { $pp['WindowStyle'] = 1 }
+
+$packageParams = @{
+  ShortcutFilePath = $pp.ShortcutFilePath
+  TargetPath = $pp.TargetPath
+  WorkingDirectory = $pp.WorkingDirectory
+  Arguments = $pp.Arguments
+  IconLocation = $pp.IconLocation
+  Description = "FreeCAD Development ${env:ChocolateyPackageVersion}"
+  WindowStyle = $pp.WindowStyle
+}
+
+if ($pp['Taskbar']) { $packageParams.Add( 'PinToTaskbar', $true ) }
+if ($pp['Admin']) {  $packageParams.Add( 'RunAsAdmin', $true ) }
+
+if ( $packageArgs.filetype -eq '7z' ) {
```

Comment:
Correct identation, code is messy. Better, create helpers so that ist is more obvious what you do here.